buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt

```diff
@@ -232,20 +249,72 @@ abstract class DackkaPlugin : Plugin<Project> {
         dependsOnAndMustRunAfter("createFullJarRelease")
 
         val dackkaFile = project.provider { project.dackkaConfig.singleFile }
-        val dackkaOutputDirectory = File(project.buildDir, "dackkaDocumentation")
 
         dackkaJarFile.set(dackkaFile)
-        outputDirectory.set(dackkaOutputDirectory)
         clientName.set(project.firebaseConfigValue { artifactId })
     }
 
-    private fun registerFiresiteTransformTask(project: Project, dackkaFilesDirectory: Provider<File>) =
-        project.tasks.register<FiresiteTransformTask>("firesiteTransform") {
-            mustRunAfter("generateDackkaDocumentation")
+    // TODO(b/248302613): Remove when dackka exposes configuration for this
+    private fun registerSeparateJavadocAndKotlinDoc(
+        project: Project,
+        dackkaOutputDirectory: Provider<File>,
+        outputDirectory: Provider<File>
+    ): TaskProvider<Task> {
+        val outputJavadocFolder = project.childFile(outputDirectory, "android")
+        val outputKotlindocFolder = project.childFile(outputDirectory, "kotlin")
+
+        val separateJavadoc = project.tasks.register<Copy>("separateJavadoc") {
+            dependsOn("generateDackkaDocumentation")
+
+            val javadocClientFolder = project.childFile(dackkaOutputDirectory, "reference/client")
+            val javadocComFolder = project.childFile(dackkaOutputDirectory, "reference/com")
+
+            fromDirectory(javadocClientFolder)
+            fromDirectory(javadocComFolder)
+
+            into(outputJavadocFolder)
+        }
+
+        val separateKotlindoc = project.tasks.register<Copy>("separateKotlindoc") {
+            dependsOn("generateDackkaDocumentation")
+
+            val kotlindocFolder = project.childFile(dackkaOutputDirectory, "reference/kotlin")
+
+            from(kotlindocFolder)
+
+            into(outputKotlindocFolder)
+        }
+
+        return project.tasks.register("separateJavadocAndKotlinDoc") {
+            dependsOn(separateJavadoc, separateKotlindoc)
+        }
+    }
+
+    private fun registerFiresiteTransformTask(
+        project: Project,
+        separatedFilesDirectory: Provider<File>,
+        targetDirectory: Provider<File>
+    ): TaskProvider<Task> {
+        val transformJavadoc = project.tasks.register<FiresiteTransformTask>("firesiteTransformJavadoc") {
+            dependsOnAndMustRunAfter("separateJavadoc")
+
+            referenceHeadTagsPath.set("en/docs/reference/android")
```

Comment:
hm, on second thought I don't think even `en/` is needed. Actually can you confirm with @rachelsaunders ?
```suggestion
            referenceHeadTagsPath.set("docs/reference/android")
```

Reply:
Also left her a chat message about this

Reply:
You should be able to use a relative reference and go all the way down to `/android/_reference-head-tags.html` and `/kotlin/_reference-head-tags.html`.

Reply:
Hm, doesn't `/android` imply "relative to the root", whatever the root is. So if we for example have a file `docs/reference/kotlin/com/google/firebase/Blah.html` with such an include what will it resolve to?

Reply:
Solved. The `root` is `google3/third_party/devsite/firebase/en/`, which makes sense as that's where `_book.yaml` and `_shared` live. I've corrected the path link as such.